src/compiler/crystal/syntax/ast.cr

```diff
@@ -1010,8 +1010,9 @@ module Crystal
     property default_value : ASTNode?
     property restriction : ASTNode?
     property doc : String?
+    property parsed_annotations : Array(Annotation)?
```

Comment:
Instead of keeping the annotations on each arg, I think we can move this to the `Def` and `Macro` objects, similar to how the splat index is kept there.

But we could do this refactor later on if we see memory issues. I just don't think that annotations on args will be that common, and so adding this overhead to every arg instead of adding just a nilable attribute to `Def` and `Macro` might be better.